src/modules/cmdpal/Microsoft.CmdPal.UI/ShellPage.xaml

```diff
@@ -22,15 +61,67 @@
             Grid.Row="0"
             VerticalAlignment="Top" />
 
-        <Grid
+        <Grid x:Name="ContentGrid"
             Grid.Row="1"
             Background="{ThemeResource LayerOnAcrylicFillColorDefaultBrush}"
             BorderBrush="{ThemeResource CardStrokeColorDefaultBrush}"
             BorderThickness="0,1,0,1">
-            <Frame Name="RootFrame" IsNavigationStackEnabled="True" />
-        </Grid>
 
+            <Grid.ColumnDefinitions>
+                <ColumnDefinition Width="3*" />
+                <ColumnDefinition x:Name="DetailsColumn" Width="Auto" />
+            </Grid.ColumnDefinitions>
+
+            <Frame Name="RootFrame" 
+                   Grid.Column="0" 
+                   IsNavigationStackEnabled="True" />
+
+            <Grid 
```

Comment:
Ah, did the original version of this PR have this encapsulated in a UserControl?

Maybe this is a larger UX narrative story in terms of what the app can do. We can sync with @niels9001 during the sync tmr maybe?

----

Like I'm wondering if we could collapse the nested grids a bit here and have just three rows (search, main content area (fill), actionbar), and then two columns (main content left, details right). Right now that details content is common, but we could maybe have different types of detail display/templates in the future? (e.g. video player?)

So, we either have all that encapsulated in a Details UserControl which knows how to handle the different types of details, or it's a Frame that then the shell swaps out to the appropriate type.

As then we'd encapsulate the show/hide and the VM stuff for that in that control, and it'd just collapse itself, which should update the layout here...